torch/optim/asgd.py

```diff
@@ -366,27 +359,23 @@ def _multi_tensor_asgd(
             torch._foreach_copy_(grouped_mus, new_mus)
             del new_mus
 
-            # update eta = lr / (1 + lambd * lr * step^alpha)
-            new_etas = torch._foreach_pow(grouped_state_steps, alpha)
-            torch._foreach_mul_(new_etas, lambd)
+            # update eta = lr / ((1 + lambd * lr * step)^alpha)
+            new_etas = torch._foreach_mul(grouped_state_steps, lambd)
             torch._foreach_mul_(new_etas, lr)
             torch._foreach_add_(new_etas, 1)
+            torch._foreach_pow_(new_etas, alpha)
```

Comment:
@janeyx99, there is an error `NotImplementedError: aten::_foreach_pow_.Scalar`. Should I try to fix it (but it's a problem in the c++ codebase, it might take some time for me to fix it), or just skip those test case?